sdk/search/Azure.Search.Documents/tests/Utilities/SharedSearchResources.cs

```diff
@@ -14,10 +15,38 @@ namespace Azure.Search.Documents.Tests
     [SetUpFixture]
     public class SharedSearchResources
     {
+        private static readonly AutoResetEvent s_event = new AutoResetEvent(true);
+        private static readonly TimeSpan s_timeout = TimeSpan.FromMinutes(1);
+
         /// <summary>
         /// The shared hotels search index.
         /// </summary>
-        public static SearchResources Search { get; set; }
+        public static SearchResources Search { get; private set; }
+
+        /// <summary>
+        /// Initializes the <see cref="Search"/> shared index if necessary.
+        /// </summary>
+        /// <param name="factory">The method to create the shared search index if necessary.</param>
+        /// <returns>A <see cref="ValueTask"/> that the method has completed.</returns>
+        public static async ValueTask EnsureInitialized(Func<Task<SearchResources>> factory)
```

Comment:
If you had the same problem with KeyVault tests, should we pull this into the shared source testing code at some point?

Reply:
The only problem was people using `??=` where thread safety was needed. No different than using `+=` etc. It's just something people need to avoid. I wouldn't be opposed to adding this to shared code, but I know @pakrym generally wants to avoid hard-to-discover helpers people may not use. But if you think this is worth it, we can make the case.

The fix for KV wasn't like this. It was simpler than that because the impact of overwriting a shared variable wasn't nearly as bad - it was just an in-memory cache of challenge scopes so we avoid lots of string parsing / allocations.

Reply:
I see - I thought it also involved the logic in this method.  That's fair then.